File: 77-combinations/77-combinations.java
class Solution {
	List<List<Integer>> ans;
	int n;
	int k;

	public List<List<Integer>> combine(int n, int k) {
		List<List<Integer>> ans = new ArrayList<>();
		if (k > n || k < 0) {
			return ans;
		}

		if (k == 0) {
			ans.add(new ArrayList<>());
			return ans;
		}
		ans = combine(n - 1, k - 1);
		ans.stream().forEach(list -> list.add(n));
		ans.addAll(combine(n - 1, k));
		return ans;
	}
}

File: 77-combinations/77-combinations.js
/**
 * @param {number} n
 * @param {number} k
 * @return {number[][]}
 */
var combine = function (n, k) {
    let findAnswer = function (n, k) {
        let ans = [];
        if (k > n || k < 0) {
            return ans;
        }
        if (k == 0) {
            ans.push([]);
            return ans;
        }
        ans = findAnswer(n - 1, k - 1);
        ans.forEach(a => a.push(n));
        let temp=findAnswer(n - 1, k);
        ans.push(...temp);
        return ans;
    }
    return findAnswer(n,k);
};
